<launch>
    <node name="controller_node" pkg="controller" type="controller_node.py" output="screen" />

    <node name="checkpoint" pkg="controller" type="checkpoint_node.py" output="screen" />

    <node name="OrientAdjustForPID" pkg="controller" type="orientation_adjustment_for_PID.py" output="screen" />


    <!-- PID Controller for thrust, required to define a input topic as thrust/setpoint
                                                                        thrust/state-->
    <node name="controller_thrust" pkg="pid" type="controller" ns="thrust" output="screen">
        <param name="Kp" value="0.0" />
        <param name="Ki" value="0.0" />
        <param name="Kd" value="0.0" />
        <param name="upper_limit" value="1" />
        <param name="lower_limit" value="-1" />
        <param name="windup_limit" value="10" />
        <param name="max_loop_frequency" value="100.0" />
        <param name="min_loop_frequency" value="100.0" />
    </node>

    <!-- PID Controller for vertical_thrust required to define a input topic as  vertical_thrust/setpoint
                                                                                                                                                     vertical_thrust/state-->
    <node name="controller_vertical_thrust" pkg="pid" type="controller" ns="vertical_thrust" output="screen">
        <param name="Kp" value="4.0" />
        <param name="Ki" value="0.018" />
        <param name="Kd" value="0.006" />
        <param name="upper_limit" value="1" />
        <param name="lower_limit" value="-1" />
        <param name="windup_limit" value="10" />
        <param name="max_loop_frequency" value="100.0" />
        <param name="min_loop_frequency" value="100.0" />
    </node>
    <!-- PID Controller for lateral_thrust required to define a input topic as  lateral_thrust/setpoint
                                                                                                                                                    lateral_thrust/state-->
    <node name="controller_lateral_thrust" pkg="pid" type="controller" ns="lateral_thrust" output="screen">
        <param name="Kp" value="0.0" />
        <param name="Ki" value="0.0" />
        <param name="Kd" value="0.0" />
        <param name="upper_limit" value="1" />
        <param name="lower_limit" value="-1" />
        <param name="windup_limit" value="10" />
        <param name="max_loop_frequency" value="100.0" />
        <param name="min_loop_frequency" value="100.0" />
    </node>

    <!-- PID Controller for orientation_Yaw required to define a input topic as  yaw/setpoint
                                                                                                                                                     yaw/state-->
    <node name="controller_yaw" pkg="pid" type="controller" ns="yaw" output="screen">
        <param name="Kp" value="0.3" />
        <param name="Ki" value="0.0" />
        <param name="Kd" value="0.0" />
        <param name="upper_limit" value="1" />
        <param name="lower_limit" value="-1" />
        <param name="windup_limit" value="10" />
        <param name="max_loop_frequency" value="100.0" />
        <param name="min_loop_frequency" value="100.0" />
    </node>

    <!--  PID Controller for pitch required to define a input topic as  pitch/setpoint
                                                                                                                                          pitch/state-->
    <!-- <node name="controller_pitch" pkg="pid" type="controller" ns="pitch" output="screen">
                                                                            <param name="Kp" value="0.0" />
                                                                            <param name="Ki" value="0.0" />
                                                                            <param name="Kd" value="0.0" />
                                                                            <param name="upper_limit" value="10" />
                                                                            <param name="lower_limit" value="-10" />
                                                                            <param name="windup_limit" value="10" />
                                                                            <param name="max_loop_frequency" value="100.0" />
                                                                            <param name="min_loop_frequency" value="100.0" />
                                                                        </node> -->
    <!-- PID Controller for roll required to define a input topic as  roll/setpoint
                                                                                                                                          roll/state-->
    <!-- <node name="controller_roll" pkg="pid" type="controller" ns="roll" output="screen">
                                                                            <param name="Kp" value="0.0" />
                                                                            <param name="Ki" value="0.0" />
                                                                            <param name="Kd" value="0.0" />
                                                                            <param name="upper_limit" value="10" />
                                                                            <param name="lower_limit" value="-10" />
                                                                            <param name="windup_limit" value="10" />
                                                                            <param name="max_loop_frequency" value="100.0" />
                                                                            <param name="min_loop_frequency" value="100.0" />
                                                                        </node> -->


</launch>
